### PR TITLE
updating sortReleases key=minor version instead of rancherMinorversion

### PR DESCRIPTION
**Pull request by slickwarren**

### CODE REVIEW

File: extensions/clusters/kubernetesversions/all.go
Comment (caliskanugur) [inline extensions/clusters/kubernetesversions/all.go:1]: Could you also update the sorting logic comment at the top of the function? I believe the second point isn't applicable anymore

File: extensions/clusters/kubernetesversions/all.go [hunk 1]
```diff
@@ -215,23 +215,19 @@ func sortReleases(releases []interface{}) (allAvailableVersions []string) {
...
-			continue
-		}
-
 		releaseKubernetesVersion, err := semver.NewVersion(strings.TrimPrefix(kubernetesVersion, "v"))
 		if err != nil {
 			continue
 		}
 
-		value, ok := availableVersionsMap[minRancherVersion.String()]
+		minor := strings.Split(releaseKubernetesVersion.Original(), ".")[1]
```
Comment (anupama2501) [inline extensions/clusters/kubernetesversions/all.go:225]: qq, what does releaseKubernetesVersion.Original() return here? can we not just use releaseKubernetesVersion?

Comment (slickwarren) [reply extensions/clusters/kubernetesversions/all.go:225]: .Original returns the string of releaseKuberenetesVersion, which is of type semver.Version. So we need the conversion to string here

File: extensions/clusters/kubernetesversions/all.go
Comment (slickwarren) [reply extensions/clusters/kubernetesversions/all.go:1]: updated the comment to be more accurate

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Review by floatingman — APPROVED** [review]
LGTM

**Review by anupama2501 — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Pull request merged**